src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/Marshalling/MarshallerHelpers.cs

```diff
@@ -332,5 +332,63 @@ public static StatementSyntax SkipInitOrDefaultInit(TypePositionInfo info, StubC
                             Token(SyntaxKind.DefaultKeyword))));
             }
         }
+
+        /// <summary>
+        /// Get the marshalling direction for a given <see cref="TypePositionInfo"/> in a given <see cref="StubCodeContext"/>.
+        /// For example, an out parameter is marshalled in the <see cref="MarshalDirection.UnmanagedToManaged"/> direction in a <see cref="MarshalDirection.ManagedToUnmanaged"/> stub,
+        /// but from <see cref="MarshalDirection.ManagedToUnmanaged"/> in a <see cref="MarshalDirection.UnmanagedToManaged"/> stub.
+        /// </summary>
+        /// <param name="info">The info for an element.</param>
+        /// <param name="context">The context for the stub.</param>
+        /// <returns>The direction the element is marshalled.</returns>
+        public static MarshalDirection GetElementMarshalDirection(TypePositionInfo info, StubCodeContext context)
```

Comment:
> GetElementMarshalDirection

The `Element` name made me think of collection elements and how we use it for `MarshalMode` (I was a bit confused looking at the call sites). Maybe just `GetMarshalDirection` (and consumers should be able to deduce that it is for the `TypePositionInfo` parameter)?